cpp/src/arrow/filesystem/filesystem_test.cc

```diff
@@ -54,6 +54,17 @@ TEST(FileInfo, BaseName) {
   ASSERT_EQ(info.base_name(), "baz.qux");
 }
 
+TEST(CalculateSSECKeyMD5, Sanity) { 
+  ASSERT_FALSE(CalculateSSECKeyMD5("").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("%^H").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("INVALID").ok());  // invalid base64
+  ASSERT_FALSE(CalculateSSECKeyMD5("MTIzNDU2Nzg5").ok());  // invalid, the input key size not match
+  // valid case
+  auto result = CalculateSSECKeyMD5("1WH9aTJ0+Tn0NLbTMHZn9aCW3Li3ViAdBsoIldPCREw=");
+  ASSERT_TRUE(result.ok());  // valid case
```

Comment:
Could you use `ASSERT_OK_AND_ASSIGN()` instead?

Reply:
done